README.Rmd

```diff
@@ -47,72 +58,82 @@ To install the development version of the package, use:
-Here we've already created some with the `include_template` being set to TRUE (the default). The helper functions below show the state of the factory.
+Here we've already created some with most of the default arguments being set to
+TRUE (the default). These default settings include both an example report and
+some associated data (`report_sources/example_report.Rmd` and 
+`data/raw/example_data.csv`).  The helper functions below show the state of the
+factory.
 
 ```{r listing}
-list_reports(factory_proj)
-list_outputs(factory_proj)
-list_deps(factory = factory_proj) # list all needed packages
+list_reports()       # list all available report sources
+list_deps()          # list all of the dependencies of the reports
+list_outputs()       # currently empty
 ```
 
 ### Step 3 - Compile report(s)
 
-#### Option 1 - A single report
-The `compile_report()` function can be used to compile a report using its name, or a partial match for its name. This is useful when you're actively working on developing a single report.
+The `compile_reports()` function can be used to compile a report using regular
+expressions matched against the full filename of reports within the factory. 
 
-```{r compile_one, eval=FALSE}
-compile_report("example_report_2019-01-31.Rmd", factory = factory_proj, quiet = TRUE)
-```
+This ability to use of regular expressions is useful when you're actively
+working on developing your reports but once the factory is setup we recommend
+passing full filenames to the function so it is always clear what will be built.
 
-Use helper functions to see progress.
-```{r compile_one_results}
-list_outputs(factory_proj)
+```{r compile}
+compile_reports( 
+  reports = "example_report.Rmd"
+)
 ```
 
-#### Option 2 - All reports with updates
-Compile all reports to get their outputs. By default, this just compiles the most recent versions.
-
-```{r compile_allupdated}
-update_reports(factory_proj)
-list_outputs(factory_proj)
+Use `list_ouputs()` to view the report outputs.
+```{r outputs}
+list_outputs()
 ```
 
-#### Option 3 - All reports
-Compile all reports to get their outputs. By default, this just compiles the most recent versions.
+`compile_reports()` can also be used to pass a set of parameters to use with a
+parameterised report (here we use a subfolder argument to distinguish the
+parameterised reports).
 
-```{r compile_all}
-update_reports(factory_proj, all = TRUE)
-list_outputs(factory_proj)
+```{r compileparams}
+compile_reports(
+  reports = "example_report.Rmd",
+  params = list(grouped_plot = FALSE),
+  subfolder = "regional"
+)
+list_outputs()
 ```
 
-### Step 4 - Consolidate latest outputs
-You can now get a single directory of the latest report outputs using:
-
-```{r shipit}
-ship_reports(factory = factory_proj, most_recent = TRUE)
-list.dirs(factory_proj)
+If you want to have an overview of your entire factory then you can use the
+`fs` package and the `dir_tree` function:
+```{r dirtree}
+fs::dir_tree()
```

Comment:
It is a bit cheeky, but would we want a wrapper e.g. `factory_overview()` for this?

Reply:
added in https://github.com/reconhub/reportfactory/pull/68/commits/63d2ad10489e459cfff67572a14f15491d0b059b